File: output/llama_4_maverick/avatar/algo_based_translation/python/java/failed_translation/atcoder_ABC143_D.java

import java.util.Arrays;
import java.util.Scanner;

public class atcoder_ABC143_D {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split("\\s+");
        int n = Integer.parseInt(input[0]);
        int[] l = new int[n];
        for (int i = 0; i < n; i++) {
            l[i] = Integer.parseInt(input[i + 1]);
        }
        Arrays.sort(l);
        int ans = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int a = l[i];
                int b = l[j];
                int right = bisectLeft(l, a + b);
                int left = bisectRight(l, Math.max(a - b, b - a));
                int tmp = Math.max(0, right - left);
                if (left <= i && i < right) tmp--;
                if (left <= j && j < right) tmp--;
                ans += tmp;
            }
        }
        System.out.println(ans / 3);
    }

    public static int bisectLeft(int[] arr, int target) {
        int left = 0;
        int right = arr.length;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (arr[mid] < target) left = mid + 1;
            else right = mid;
        }
        return left;
    }

    public static int bisectRight(int[] arr, int target) {
        int left = 0;
        int right = arr.length;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (arr[mid] <= target) left = mid + 1;
            else right = mid;
        }
        return left;
    }
}

File: output/deepseek_chat/avatar/algo_based_translation/java/python/successful_translation/atcoder_ABC143_D.py

import sys
import bisect

def main():
    data = sys.stdin.read().split()
    n = int(data[0])
    L = list(map(int, data[1:1+n]))
    L.sort()
    count = 0
    for i in range(n):
        for j in range(i+1, n):
            a = L[i]
            b = L[j]
            target = a + b
            left = j + 1
            right = n - 1
            if left > right:
                continue
            pos = bisect.bisect_left(L, target, left, n)
            count += (pos - left)
    print(count)

if __name__ == "__main__":
    main()

